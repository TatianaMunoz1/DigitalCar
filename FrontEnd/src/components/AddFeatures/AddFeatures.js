import React, { useState, useEffect } from "react";
import styles from "./AddFeatures.module.css";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faPlus, faTimes } from "@fortawesome/free-solid-svg-icons";
import Swal from "sweetalert2";
import FeaturePicker from "../FeaturePicker/FeaturePicker";

function AddFeatures(props) {
  const [listFeatures, setListFeatues] = useState(null);
  const [validation, setValidation] = useState(false);
  const [newCarFeatures, setNewCarFeatures] = useState([]);
  const [newCarFeature, setNewCarFeature] = useState({
    name: "",
    icon: "",
  });
  const [featureValidation, setFeatureValidation] = useState({
    name: null,
    icon: null,
  });

  useEffect(() => {
    props.handleAddNewFeatures([...newCarFeatures]);
  }, [newCarFeatures]);

  const handleAddFeature = () => {
    if (featureValidation.name === true && featureValidation.icon === true) {
      setValidation(true);
      setNewCarFeatures([...newCarFeatures, newCarFeature]);
      setNewCarFeature({
        name: "",
        icon: "",
      });
    } else {
      setValidation(false);
    }
  };

  const handleFeaturePicker = (feature) => {
    if (feature !== null && feature !== undefined) {
      setNewCarFeature({
        name: feature.name,
        icon: feature.icon,
      });
      setNewCarFeatures([
        ...newCarFeatures,
        {
          name: feature.name,
          icon: feature.icon,
        },
      ]);
      setValidation(true);
    }
  };

  const handleDeleteFeature = (e) => {
    Swal.fire({
      title: "Â¿Desea borrar el atributo?",
      icon: "warning",
      showCancelButton: true,
      background: "#f3f3f3",
      confirmButtonColor: "#b40505",
      cancelButtonColor: "#000000",
      confirmButtonText: "Borrar",
      cancelButtonText: "No borrar",
    }).then((result) => {
      if (result.isConfirmed) {
        const idImage = e.target.id;
        const array = [...newCarFeatures];
        array.splice(idImage, 1);
        setNewCarFeatures([...array]);
      }
    });
  };

  useEffect(() => {
    const listFeature = () =>
      newCarFeatures.map((carFeature, index) => (
        <div
          key={`${carFeature.name}_${index}`}
          className={styles.new_feature_box}
        >
          <div className={styles.new_feature}>
            <input
              className={styles.input}
              type="text"
              value={carFeature.name}
              disabled
            />
          </div>
          <div className={styles.btn_new_feature_container}>
            <button
              className={styles.btn_delete_feature}
              type="button"
              id={index}
              onClick={(e) => handleDeleteFeature(e)}
            >
              <FontAwesomeIcon icon={faTimes} />
            </button>
          </div>
        </div>
      ));
    setListFeatues(listFeature);
  }, [newCarFeatures]);

  return (
    <div className={styles.features}>
      <div className={styles.new_features}>
        {newCarFeatures.length > 0 && <div>{listFeatures}</div>}
      </div>

      <div className={styles.add_features_section}>
        <div className={styles.add_features}>
          <FeaturePicker handleFeaturePicker={handleFeaturePicker} />
        </div>
        <div className={styles.btn_container}>
          <button
            className={styles.btn_add_feature}
            type="button"
            onClick={handleAddFeature}
          >
            <FontAwesomeIcon icon={faPlus} />
          </button>
        </div>
      </div>
    </div>
  );
}

export default AddFeatures;
