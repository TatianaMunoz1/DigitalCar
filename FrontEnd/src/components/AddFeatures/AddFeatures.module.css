.input {
  height: 30px;
  width: 100%;
  padding: 6px;
  font-size: 14px;
  background: #ffffff;
  box-shadow: 0px 1px 5px rgba(0, 0, 0, 0.15);
  border: none;
  border-radius: 5px;
}
.add_features_section,
.new_feature_box {
  background: #e9e9e9;
  box-shadow: 2px 2px 5px rgba(0, 0, 0, 0.15);
  display: flex;
  width: 100%;
  gap: 10px;
  padding: 20px 10px;
  margin-bottom: 12px;
  align-items: center;
}
.new_features {
  padding: 15px 0px;
}
.new_feature,
.add_features {
  display: flex;
  flex-direction: column;
  gap: 20px;
  width: 100%;
}
.new_feature_box:hover {
  background: #e0e0e0;
  transition: all 0.3s ease-out;
}
.btn_add_feature,
.btn_delete_feature {
  height: 30px;
  width: 30px;
  border-radius: 100%;
  font-size: 15px;
  line-height: 30px;
  padding: 0px;
  background: #ff7e19;
  box-shadow: 1px 1px 5px rgba(0, 0, 0, 0.15);
  color: #fff;
  border: none;
  cursor: pointer;
}
.btn_delete_feature {
  background: #000000;
}
.btn_delete_feature:hover {
  background: #7c7c7c;
  transition: all 0.3s ease-out;
}
.btn_delete_feature:hover {
  background: #7c7c7c;
  transition: all 0.3s ease-out;
}
.btn_add_feature:hover {
  background: #ffa25a;
  transition: all 0.3s ease-out;
}
.btn_new_feature_container,
.btn_container {
  width: 65px;
  display: flex;
  justify-content: right;
  align-self: center;
}

/*-----TABLET-----*/
@media (min-width: 760px) and (max-width: 1280px) {
  .input {
    height: 40px;
  }
  .add_features,
  .new_feature {
    display: grid;
    grid-template-rows: 1fr 1fr;
    gap: 20px;
    width: 100%;
  }
  .add_features_section,
  .new_feature_box {
    background: #e9e9e9;
    box-shadow: 2px 2px 5px rgba(0, 0, 0, 0.15);
    display: flex;
    width: 100%;
    height: 100%;
    gap: 20px;
    padding: 25px;
    margin-bottom: 12px;
    align-items: center;
  }
  .btn_add_feature,
  .btn_delete_feature {
    height: 40px;
    width: 40px;
    border-radius: 100%;
    font-size: 20px;
    line-height: 40px;
    padding: 0px;
    border: none;
    cursor: pointer;
  }
  .btn_new_feature_container,
  .btn_container {
    width: 85px;
    display: flex;
    justify-content: right;
    align-items: center;
  }
}

/*-----DESKTOP-----*/
@media (min-width: 1281px) {
  .input {
    height: 40px;
  }
  .add_features,
  .new_feature {
    display: grid;
    grid-template-rows: 1fr;
    grid-template-columns: 8fr 2fr;
    gap: 20px;
    width: 100%;
  }
  .new_feature_box,
  .add_features_section {
    padding: 30px;
  }
  .btn_add_feature,
  .btn_delete_feature {
    height: 40px;
    width: 40px;
    border-radius: 100%;
    font-size: 20px;
    line-height: 40px;
    align-self: flex-end;
    cursor: pointer;
  }
  .btn_new_feature_container,
  .btn_container {
    width: 85px;
    display: flex;
    justify-content: right;
    align-items: center;
  }
  .btn_container {
    align-self: flex-end;
  }
}
