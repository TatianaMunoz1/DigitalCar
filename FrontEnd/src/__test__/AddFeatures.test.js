import React, { useEffect } from "react";
import { shallow, mount, render, mock } from "enzyme";
import { Router } from "react-router-dom";
import AddFeatures from "../components/AddFeatures/AddFeatures";

describe("add features", () => {
    test("render add features ", () => {
        const wrapper = shallow(<AddFeatures/>)
        expect(wrapper).toMatchSnapshot();
    });

    test('should call handleInputFeatureChange method', () => {
        const onClickSpy = jest.fn();
        const wrapper = shallow(<AddFeatures handleInputFeatureChange={onClickSpy} />);
        expect(wrapper).toMatchSnapshot();
        wrapper.find('button').simulate('click');
        expect(onClickSpy).toHaveBeenCalled();
});

});