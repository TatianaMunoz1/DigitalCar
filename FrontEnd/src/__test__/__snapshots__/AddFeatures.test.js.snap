// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`add features render add features  1`] = `ShallowWrapper {}`;

exports[`add features should call handleInputFeatureChange method 1`] = `ShallowWrapper {}`;
